### ISSUE TITLE
Inconsistency of io_shape_dict in generated driver.py

### ISSUE THREAD

**Issue by oskarnatan** [Xilinx/finn#334]
Hello,

Currently, I am developing a semantic segmentation model with the input shape of NHWC = 1x32x32x3 and the output shape of 1x32x32x23 (23 class). I have done a lot of experiments with plenty of model variants to be built with FINN's example end-to-end notebook. However, I see that the output 'io_shape_dict ' in generated file driver.py in "pynq_deployment_xxxxxxxxx" folder is inconsistent. For example, I have 2 simple models below:

Model A: input -> qnn.QuantConv2d -> nn.Sigmoid -> output
Model B: input -> 4x(qnn.QuantConv2d + nn.BatchNorm + nn.ReLU) -> qnn.QuantConv2d -> nn.Sigmoid -> output

Both models have the same I/O size but they have different 'io_shape_dict' in each driver.py file.
Model A:
```
# dictionary describing the I/O of the FINN-generated accelerator
io_shape_dict = {
    # FINN DataType for input and output tensors
    "idt" : DataType.UINT8,
    "odt" : DataType.INT16,
    # shapes for input and output tensors (NHWC layout)
    "ishape_normal" : (1, 32, 32, 3),
    "oshape_normal" : (1, 32, 32, 23),
    # folded / packed shapes below depend on idt/odt and input/output
    # PE/SIMD parallelization settings -- these are calculated by the
    # FINN compiler.
    "ishape_folded" : (1, 32, 32, 3, 1),
    "oshape_folded" : (1, 32, 32, 23, 1),
    "ishape_packed" : (1, 32, 32, 3, 1),
    "oshape_packed" : (1, 32, 32, 23, 2),
    "input_dma_name" : 'idma0',
    "number_of_external_weights": 0
}
```
Model B:
```
# dictionary describing the I/O of the FINN-generated accelerator
io_shape_dict = {
    # FINN DataType for input and output tensors
    "idt" : DataType.UINT8,
    "odt" : DataType.INT16,
    # shapes for input and output tensors (NHWC layout)
    "ishape_normal" : (1, 32, 32, 3),
    "oshape_normal" : (1, 32, 32, 1),
    # folded / packed shapes below depend on idt/odt and input/output
    # PE/SIMD parallelization settings -- these are calculated by the
    # FINN compiler.
    "ishape_folded" : (1, 32, 32, 1, 3),
    "oshape_folded" : (1, 32, 32, 1, 1),
    "ishape_packed" : (1, 32, 32, 1, 3),
    "oshape_packed" : (1, 32, 32, 1, 2),
    "input_dma_name" : 'idma0',
    "number_of_external_weights": 0
}
```

As can be seen from both 'io_shape_dict', they have different output shapes (oshape). After I run the 'validate.py' file (also generated in "pynq_deployment_xxxxxxxxx" folder), the model's output shape is following their "oshape_packed".

So, how could this possible? the model's output is expected to be 1x32x32x23, but it has a different output shape based on their generated driver.py's io_shape_dict. Thank you.

**Comment by fpjentzsch** [comment]
Cause has been found, refer to gitter channel for further discussion: https://gitter.im/xilinx-finn/community?at=60b43be5688a2624b8bf8d4f
